dags/hindawi/hindawi_api_client.py

```diff
@@ -10,12 +10,12 @@ def __init__(
         base_url=None,
         path_segments=None,
     ):
-        self.base_url = base_url or getenv(
-            "HINDAWI_API_BASE_URL", "https://www.hindawi.com"
-        )
-        self.path_segments = path_segments or ["oai-pmh", "oai.aspx"]
+        self.base_url = base_url
+        self.path_segments = path_segments
 
     def get_articles_metadata(self, parameters, doi=None):
+        self.base_url = getenv("HINDAWI_API_BASE_URL", "https://www.hindawi.com")
```

Comment:
why you override here the `self.base_url`? shouldn't be this logic be in `__init__`?